applications/dashboard/controllers/Api/RolesApiController.php

```diff
@@ -0,0 +1,288 @@
+<?php
+/**
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license GPLv2
+ */
+
+use Garden\Schema\Schema;
+use Garden\Web\Data;
+use Garden\Web\Exception\NotFoundException;
+use Garden\Web\Exception\ServerException;
+use Vanilla\Utility\CapitalCaseScheme;
+
+/**
+ * API Controller for the `/roles` resource.
+ */
+class RolesApiController extends AbstractApiController {
+
+    /** @var CapitalCaseScheme */
+    private $caseScheme;
+
+    /** @var Schema */
+    private $idParamSchema;
+
+    /** @var PermissionModel */
+    private $permissionModel;
+
+    /** @var RoleModel */
+    private $roleModel;
+
+    /** @var Schema */
+    private $rolePostSchema;
+
+    /** @var Schema */
+    private $roleSchema;
+
+    /**
+     * RolesApiController constructor.
+     *
+     * @param RoleModel $roleModel
+     */
+    public function __construct(RoleModel $roleModel, PermissionModel $permissionModel) {
+        $this->roleModel = $roleModel;
+        $this->permissionModel = $permissionModel;
+        $this->caseScheme = new CapitalCaseScheme();
+    }
+
+    /**
+     * Delete a role.
+     *
+     * @param int $id The ID of the role.
+     * @return null
+     */
+    public function delete($id) {
+        $this->permission('Garden.Settings.Manage');
+
+        $in = $this->idParamSchema()->setDescription('Delete a role.');
+        $out = $this->schema([], 'out');
+
+        $this->roleByID($id);
+        $this->roleModel->deleteID($id);
+
+        return null;
```

Comment:
I don't think it improves readability. On the contrary if I see return null at the end of a function my first reaction would be to check for another return statement in that function since the default return is null.